grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -474,11 +476,14 @@ class UserService {
     }
 
     def getDashboardConfig(User user, String id) {
-        def fullConfig = grailsApplication.config.openboxes.dashboardConfig
-        def mainDashboardId = grailsApplication.config.openboxes.dashboardConfig.mainDashboardId
+        Gson gson = new Gson()
+        def fullConfig = gson.fromJson(
```

Comment:
Is there a more elegant way to achieve this? If the `.clone()` does not work, and there is a need to do a deep copy.

Reply:
I didn't find any better solution for creating a deep copy, this one looks better for me, because we are currently using the gson library so I didn't have to install a new one.
Mainly I based on this article: https://www.baeldung.com/java-deep-copy

Reply:
@alannadolny ok, please add a comment above this line about what is happening here and link this article too.